test/set_test.js

```diff
@@ -0,0 +1,43 @@
+const proxyquire = require('proxyquire').noCallThru();
+const expect     = require('chai').expect;
+const msg        = require('./stubs/message.js');
+const SetCommand = proxyquire(
+    '../api/commands/reminders/set.js',
+    { 
+        'discord.js-commando': require('./stubs/Command.js'),
+        '@global': true
+    }
+);
+
+describe('#run', () => {
+    const content = "Hello World!";
+    const target  = { id: 1 };
+
+    describe('time parsing', () => {
+        context('when date format is incorrect', () => {
+            let time = 'hello';
+
+            it('throws invalid format exception', () => {
+                expect(
+                    new SetCommand({}).run(
+                        msg, 
+                        { target: target, content: content, datetime: time }
+                    )
+                ).to.match(/view accepted datetime formats/);
+            });
+        });
+
+        context('when date is in the past', () => {
+            let time = 'yesterday at noon';
+
+            it('throws time pasted exception', () => {
+                expect(
+                    new SetCommand({}).run(
+                        msg, 
+                        { target: target, content: content, datetime: time }
+                    )
+                ).to.eq('Error! You cannot schedule a reminder for the past.');

```

Comment:
Why `to.match()` above but `to.eq()` here?

Reply:
message was ridiculously long in the above one so match there
eq for the shorter one.

I prefer eq over match for both performance as well as clarity, but that's one long err message

Reply:
Since `set.js` has an error messages object, could we not just refer to the err that way? Essentially, something like `to.eq(SetCommand.errors.pastTime)`.

Reply:
I dont think `errors` is a property on SetCommand which is a class defined with just a `constructor` and `run` method

looks like it's a `const` on the module itself. cant remember node encapsulation too well but I'd guess it isn't exposed. Doesn't mean I can't access it, but if it isn't exposed the solution would be even more messy.

Reply:
What do you think about moving that object out into an `error_messages.json`? Rather than create individual error objects within future new commands.

Reply:
👍

Reply:
within scope of this pr or a separate?

Reply:
Your call.